cargo-dist/src/platform.rs

```diff
@@ -0,0 +1,261 @@
+    MinMuslVersion { major: u64, series: u64 },
+    Rosetta2,
+}
+
+#[derive(Debug, Clone, Default)]
+pub struct PlatformSupport {
+    pub archives: Vec<FetchableArchive>,
+    pub platforms: SortedMap<TargetTriple, Vec<PlatformEntry>>,
+}
+
+#[derive(Debug, Clone)]
+pub struct FetchableArchive {
+    pub id: ArtifactId,
+    pub native_runtime_conditions: Vec<RuntimeCondition>,
+    pub target_triples: Vec<TargetTriple>,
+    pub sha256sum: Option<String>,
+    pub binaries: Vec<String>,
+    pub zip_style: ZipStyle,
+}
+
+pub type FetchableArchiveIdx = usize;
+
+#[derive(Debug, Clone)]
+pub struct PlatformEntry {
+    pub quality: SupportQuality,
+    pub runtime_conditions: Vec<RuntimeCondition>,
+    pub archive_idx: FetchableArchiveIdx,
+}
+
+impl PlatformSupport {
+    pub(crate) fn new(dist: &DistGraphBuilder, release_idx: ReleaseIdx) -> PlatformSupport {
+        let mut platforms = SortedMap::<TargetTriple, Vec<PlatformEntry>>::new();
+        let release = dist.release(release_idx);
+        let mut archives = vec![];
+
+        // Gather up all the fetchable archives
+        for &variant_idx in &release.variants {
+            // Compute the artifact zip this variant *would* make *if* it were built
+            // FIXME: this is a kind of hacky workaround for the fact that we don't have a good
+            // way to add artifacts to the graph and then say "ok but don't build it".
+            let (artifact, binaries) =
+                dist.make_executable_zip_for_variant(release_idx, variant_idx);
+            let archive = FetchableArchive {
+                id: artifact.id,
+                target_triples: artifact.target_triples,
+                binaries: binaries
+                    .into_iter()
+                    .map(|(_, dest_path)| dest_path.file_name().unwrap().to_owned())
+                    .collect(),
+                zip_style: artifact.archive.as_ref().unwrap().zip_style,
+                sha256sum: None,
+                native_runtime_conditions: vec![],
+            };
+            archives.push(archive);
+        }
+
+        // Compute what platforms each archive Really supports
+        for (archive_idx, archive) in archives.iter().enumerate() {
+            let supports = supports(archive_idx, archive);
+            for (target, support) in supports {
+                platforms.entry(target).or_default().push(support);
+            }
+        }
+
+        // Now sort the platform-support so the best options come first
+        for (_platform, support) in &mut platforms {
+            support.sort_by(|a, b| {
+                // Sort by SupportQuality, tie break by artifact name (for stability)
+                a.quality.cmp(&b.quality).then_with(|| {
+                    let archive_a = &archives[a.archive_idx];
+                    let archive_b = &archives[b.archive_idx];
+                    archive_a.id.cmp(&archive_b.id)
+                })
+            });
+        }
```

Comment:
This is the purpose of the SupportQuality thing. Oh right, providing all options for a platform would also theoretically allow an installer to say "there's a sickrad option, but i can't check for some RuntimeConditions it requires, so i'm gonna use a worse one i'm confident in".

Nice to have the option.